cpp/include/cudf/filling.hpp

```diff
@@ -203,5 +203,33 @@ std::unique_ptr<column> sequence(
   scalar const& init,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Generate a sequence of timestamps beginning at `init` and incrementing by `months` for each successive element, i.e., `output[i] = init + i * months` for `i` in `[0, size)`.
+ *
+ * If a given date is invalid, the date is scaled back to the last available day of that month.
+ *
+ * Example:
+ * ```
+ * size = 3
+ * init = 2020-01-31 08:00:00
+ * months = 1
+ * return = [2020-01-31 08:00:00, 2020-02-29 08:00:00, 2020-03-31 08:00:00]
+ * ```
+ *
+ * @throw cudf::logic_error if input datatype is not a TIMESTAMP
+ *
+ * @param size Number of timestamps to generate
+ * @param init The initial timestamp
+ * @param months Months to increment
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ *
+ * @returns Timestamps column with sequences of months.
+ */
+std::unique_ptr<cudf::column> calendrical_month_sequence(
+  size_type size,
+  scalar const& init,
+  size_type months,
```

Comment:
Would it be too bold to suggest this type be made a `cuda::std::months`? What that wreak havoc on Cython needing libcu++? There will come a day soon where we do need to depend on libcu++ externally, but it doesn't need to be this day.

Reply:
@shwina can speak more. But the fact that `libcpp.chrono` is [non-existent](https://github.com/cython/cython/tree/master/Cython/Includes/libcpp) in cython probably suggest there's still some way till it's supported.

Reply:
Are we tabling this for future work or where did we land?

Reply:
This sounds like we would need `libcu++` Cython bindings. That definitely seems like something to table (at least for this PR).

Reply:
Agree, we'd need to add basically mimic [Cython's libcpp bindings](https://github.com/cython/cython/tree/master/Cython/Includes/libcpp). We could use those as a starting point, but I don't even think that's cuDF-specific and should probably live elsewhere.